addons/mail/models/mail_thread.py

```diff
@@ -1428,7 +1428,7 @@ def _message_route_process(self, message, message_dict, routes):
                 # postponed after message_post, because this is an external message and we don't want to create
                 # duplicate emails due to notifications
                 new_msg.write({'partner_ids': original_partner_ids})
-        return thread_id
+        return thread.sudo(self.env.su)
```

Comment:
Actually maybe just
```suggestion
        return thread.with_env(self.env)
```
and avoid any surpise

I'll move the branch to our repo to have matching enterprise branch anyway so guess I'll do it.
Thanks for the PR